src/FixedDataTableCellGroup.js

```diff
@@ -76,25 +87,70 @@ var FixedDataTableCellGroupImpl = React.createClass({
     }, false);
 
     var currentPosition = 0;
-    for (var i = 0, j = columns.length; i < j; i++) {
+    var count = 0;
+
+    var newColumnsToRender = new Array(props.maxVisibleColumns);
+    var positions = new Array(props.maxVisibleColumns);
+
+    for (var i = 0; i < columns.length; i++) {
       var columnProps = columns[i].props;
       var recycable = columnProps.allowCellsRecycling && !isColumnReordering;
       if (!recycable || (
-            currentPosition - props.left <= props.width &&
-            currentPosition - props.left + columnProps.width >= 0)) {
-        var key = columnProps.columnKey || 'cell_' + i;
-        cells[i] = this._renderCell(
-          props.rowIndex,
-          props.rowHeight,
-          columnProps,
-          currentPosition,
-          key,
-          contentWidth,
-          isColumnReordering
-        );
+          currentPosition - props.left <= props.width &&
+          currentPosition - props.left + columnProps.width >= 0)) {
+        positions[i] = currentPosition;
+        newColumnsToRender[count++] = i;
       }
+
       currentPosition += columnProps.width;
     }
+
+    //TODO move this recycle logic into main state
+    const newColumnsSet = new Set(newColumnsToRender);
```

Comment:
How does this logic hold up if columns are added or removed?  Let's be sure to test against the hide / remove column example.